### PR TITLE
Fix the middle end for OCaml 5 minus

### PR DESCRIPTION
**Pull request by mshinwell**
Not yet ready for review.

Based on #1949

### CODE REVIEW

**Comment by mshinwell** [comment]
This is now ready for review, just the "Fixing middle end" commit.  Have been a bit conservative with some of the `Mutable` annotations on `Pfield` probably, but this code will be deleted soon anyway...

**Comment by mshinwell** [comment]
I've asked @Ekdohibs if she can review.

**Comment by mshinwell** [comment]
(Cmm-related parts still to come)

File: middle_end/flambda/lift_let_to_initialize_symbol.ml [hunk 1]
```diff
@@ -277,7 +277,9 @@ let rec make_named (symbol, (path:access list)) : Flambda.named =
     Expr (
       Flambda.create_let block (make_named (symbol, t))
         (Flambda.create_let field
-           (Prim (Pfield (h, Pvalue Pgenval), [block], Debuginfo.none))
+           (* CR mshinwell: double-check this change *)
+           (Prim (Pfield (h, Pvalue Pgenval, Pointer, Mutable), [block],
+             Debuginfo.none))
```
Comment (chambart) [inline middle_end/flambda/lift_let_to_initialize_symbol.ml:282]: This seems safe: Values contained in the symbol can be anything.
But the read block is actually immutable.

File: middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml [hunk 2]
```diff
@@ -1085,7 +1085,8 @@ let convert_lprim ~big_endian (prim : L.primitive) (args : Simple.t list list)
       Simple.const (Reg_width_const.tagged_immediate (Targetint_31_63.of_int n))
     in
     [Binary (Int_arith (I.Tagged_immediate, Add), arg, Simple const)]
-  | Pfield (index, sem), [[arg]] ->
+  | Pfield (index, _int_or_ptr, sem), [[arg]] ->
+    (* CR mshinwell: make use of the int-or-ptr flag (new in OCaml 5)? *)
```
Comment (chambart) [inline middle_end/flambda2/from_lambda/lambda_to_flambda_primitives.ml:1089]: We could use it. It's not obvious that this is very useful.

**Comment by chambart** [comment]
Commit [Fixing middle end ](https://github.com/ocaml-flambda/flambda-backend/pull/1958/commits/2bb795fc2131c4497f10e0dc24b83da7578649b5) looks good

File: middle_end/flambda/lift_let_to_initialize_symbol.ml (hunk 1 above)
Comment (mshinwell) [reply middle_end/flambda/lift_let_to_initialize_symbol.ml:282]: Fixed

**Comment by mshinwell** [comment]
Necessary changesets included in PR1949

**Pull request closed without merging**